File: code_samples/scala/security_authorization.scala
/**
 * Security: Authorization
 * AI/ML Training Sample
 */

class Authorization {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object AuthorizationExample {
  def main(args: Array[String]): Unit = {
    val instance = new Authorization()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}

File: code_samples/haskell/security_authorization.hs
-- Security: Authorization
-- AI/ML Training Sample

module Authorization where

data Authorization = Authorization {
    getData :: String
} deriving (Show, Eq)

process :: Authorization -> String -> Authorization
process obj input = obj { getData = input }

validate :: Authorization -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = Authorization { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)
